be/src/vec/functions/function_string.h

```diff
@@ -553,36 +560,125 @@ class FunctionStringConcatWs : public IFunction {
         fmt::memory_buffer buffer;
         std::vector<std::string_view> views;
 
-        for (size_t i = 0; i < input_rows_count; ++i) {
-            auto& seq_offsets = *offsets_list[0];
-            auto& seq_chars = *chars_list[0];
-            auto& seq_nullmap = *null_list[0];
-            if (seq_nullmap[i]) {
-                res_data.push_back('\0');
-                res_offset[i] = res_data.size();
-                continue;
+        DataTypePtr array_column_type =
+                is_array(remove_nullable(block.get_by_position(arguments[1]).type))
+                        ? remove_nullable(block.get_by_position(arguments[1]).type)
+                        : nullptr;
+        if (array_column_type) {
```

Comment:
since this 'execute_impl' function is too long, it is recommended to split sub-functions.

Reply:
done